lib/internal/modules/esm/worker.js

```diff
@@ -164,24 +178,39 @@ async function customizedModuleWorker(lock, syncCommPort, errorHandler) {
     setImmediate(() => {});
   });
 
+  const allThreadRegisteredHandlerPorts = [];
+  function registerHandler(toWorkerThread) {
+    toWorkerThread.on('message', handleMessage);
+    allThreadRegisteredHandlerPorts.push(toWorkerThread);
```

Comment:
```suggestion
    ArrayPrototypePush(allThreadRegisteredHandlerPorts, toWorkerThread);
```

Reply:
This is the worker, which I think we said does not need to use primordials?

Reply:
IIUC this is run as part of the message handler, so we'd want to protect it from user code mutating globals. THe part where primordials are not important would be for code that runs before user code.

Reply:
Ahh, yes. That is what it was, thanks! I'll subsequently add a note to this file to that effect.